### DISCUSSION TITLE
OpenGLControlBase on Linux (Raspberry PI 4)

### DISCUSSION THREAD

**Discussion by mblack101** [AvaloniaUI/Avalonia#8648] [General]
I wrote a small app in Avalonia that uses OpenGLControlBase to so I can use OpenGL inside a control and finally got it to work after UseWgl=true was set. Now I'm trying to run it on my raspberry pi 4 but I just get a blank canvas instead.

Is there any setting that must be enabled for OpenGLControlBase to use OpenGL in Linux? I previously installed the latest OpenGL libraries on my pi.

Thanks,

- Mike

**Comment by kekekeks** [comment]
Make sure that you are using MESA drivers

**Reply by mblack101** [reply]
I can try the one at https://docs.mesa3d.org/drivers/vc4.html to see if that works. Do I also have to change the following?

    public static AppBuilder BuildAvaloniaApp()
            => AppBuilder.Configure<App>()
                .With(new Win32PlatformOptions { UseWgl = true })
                .UsePlatformDetect()
                .LogToTrace();
        }
